renderlib/MathUtil.cpp

```diff
@@ -33,3 +33,11 @@ AffineSpace3f::inverse() const
   glm::vec3 ip = -xfmVector(il, p);
   return AffineSpace3f(il, ip);
 }
+
+AffineSpace3f::AffineSpace3f(const glm::quat& orientation, const glm::vec3& p)
+  : p(p)
+{
+  // convert quat to linspace:
+  glm::mat3 m = glm::mat3_cast(orientation);
+  l = LinearSpace3f(m[0], m[1], m[2]);
+}
```

Comment:
Does `l`/`LinearSpace3f` describe a rotation matrix?

Reply:
yes, basically it's a 3x3 matrix (or set of 3 basis vectors) but it doesn't strictly have to be a rotation.  In all cases I'm using it, it is a rotation e.g. between camera and world space